### DISCUSSION TITLE
Elevate empty waymarks

### DISCUSSION THREAD

**Discussion by muromir** [mapsforge/vtm#1055] [Q&A] [answered]
Hello,
I would like to ask if someone noticed and has workaround on below issue:
using VTM library with osmandoromaps some waymarks are empty as below on picture 
![image](https://github.com/mapsforge/vtm/assets/104918108/cd32b6c8-953b-4323-9ecf-863eec19b69e)

I cannot observe this behavior using mapsforge library with the same themes and maps 
![image](https://github.com/mapsforge/vtm/assets/104918108/6196213b-4b5d-47c0-b9d6-d8783cb24ec0)

It doesnt depend on zoom
Did i miss some configuring or it is a bug 

Thanks

**Comment by devemux86** [comment]
@muromir what map theme?

Are these symbols rendered from image files?

**Comment by devemux86** [comment]
VTM engine is different from the Mapsforge engine and VTM has more advanced theme rules.
VTM is compatible with all Mapsforge maps.

OpenAndroMaps themes like Elevate are Mapsforge themes.
VTM can read them, but probably not all Mapsforge theme rules can appear the same.

Nevertheless, the symbols should appear the same, as both use the same SVG reader.

**Reply by muromir** [reply]
Thanks @devemux86 

I use Elevate.xml themes and filter using elv-hiking.

loadTheme (styleId,"elv-hiking");

I didnt changed anything in themes only downloaded from OAM. 

"_Nevertheless, the symbols should appear the same, as both use the same SVG  #reader._"

exactly I expected the same

**Reply by devemux86** [reply]
Can you find the line of this rule in the Elevate theme?

**Reply by muromir** [reply]
Yes 
Elevate.xml starting from line 6944 as below:

<rule cat="waymarks" e="any" k="information" v="guidepost" zoom-min="15">
		<symbol id="s_guidepost" src="file:ele-res/s_guidepost.svg" symbol-height="16" />
		<rule e="any" k="*" v="*">
			<caption k="addr:housenumber" priority="15" symbol-id="s_guidepost" display="always" position="right" font-size="10" font-style="bold" fill="#FFFFFF" stroke="#333333" stroke-width="2" />
		</rule>
	</rule>
	<rule cat="waymarks" e="node" k="osmc" v="osmc_yes">
		<rule e="node" k="*" v="*" zoom-max="14" zoom-min="12">
			<rule e="node" k="osmc_background" v="wmbg_black|wmbg_black_circle|wmbg_black_frame|wmbg_black_round">
				<symbol src="file:ele-res/wmbg_black.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_blue|wmbg_blue_circle|wmbg_blue_frame|wmbg_blue_round">
				<symbol src="file:ele-res/wmbg_blue.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_brown|wmbg_brown_circle|wmbg_brown_frame|wmbg_brown_round">
				<symbol src="file:ele-res/wmbg_brown.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_green|wmbg_green_circle|wmbg_green_frame|wmbg_green_round">
				<symbol src="file:ele-res/wmbg_green.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_orange|wmbg_orange_circle|wmbg_orange_frame|wmbg_orange_round">
				<symbol src="file:ele-res/wmbg_orange.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_purple|wmbg_purple_circle|wmbg_purple_frame|wmbg_purple_round">
				<symbol src="file:ele-res/wmbg_purple.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_red|wmbg_red_circle|wmbg_red_frame|wmbg_red_round">
				<symbol src="file:ele-res/wmbg_red.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_white|wmbg_white_circle|wmbg_white_frame|wmbg_white_round|~">
				<symbol src="file:ele-res/wmbg_white.svg" symbol-height="6" />
			</rule>
			<rule e="node" k="osmc_background" v="wmbg_yellow|wmbg_yellow_circle|wmbg_yellow_frame|wmbg_yellow_round">
				<symbol src="file:ele-res/wmbg_yellow.svg" symbol-height="6" />
			</rule>
		</rule>

**Reply by muromir** [reply]
![image](https://github.com/mapsforge/vtm/assets/104918108/9caeaa79-5292-43f1-a73a-e6f1c81adab2)

**Reply by devemux86** [reply]
But what symbol is this? What rule?

Line symbols are converted, maybe they cannot work correctly.
Node symbols should work.

Note: Elevate theme is created & tested on Mapsforge engine.

**Reply by muromir** [reply]
I am not sure if I can correctly answer your question as i am not familiar with OSM themes but i believe it is below rule:

```
<rule e="node" k="osmc_foreground" v="wmfg_green_dot">
				<symbol src="file:ele-res/wmfg_green_dot.svg" display="always" symbol-height="12" />
			</rule>
``` 
because in VTM is missing exactly  wmfg_green_dot.svg in frame when you compare both pictures. The same is for wmfg_green_bar.svg.

```
<rule e="node" k="osmc_foreground" v="wmfg_green_bar">
				<symbol src="file:ele-res/wmfg_green_bar.svg" display="always" symbol-height="12" />
			</rule>
```
I found much more signs are not displayed as for example  wmfg_red_bar.svg end so on... 
```
<rule e="node" k="osmc_foreground" v="wmfg_red_bar">
				<symbol src="file:ele-res/wmfg_red_bar.svg" display="always" symbol-height="12" />
			</rule>
```

**Reply by devemux86** [reply, marked as answer]
Elevate theme seems to use multiple stacked symbols to render 1 icon at the same location.
Look for "osmc_background" and "osmc_foreground" rules and "wmbg_" and "wmfg_" icons.

This cannot work on OpenGL, we need to use 1 symbol.

Also it is not good for performance in all engines to render separately so many elements. For example:
- https://github.com/mapsforge/mapsforge/issues/1437

**Reply by muromir** [reply]
Many thanks @devemux86 for explanation so I think this issue can be direct to OAM developer
